app/services/katello/pulp3/api/content_guard.rb

```diff
@@ -20,19 +22,50 @@ def rhsm_api_client
           PulpCertguardClient::ContentguardsRhsmApi.new(api_client)
         end
 
-        def create(name = "RHSMCertGuard", ca_certificate = Cert::Certs.ca_cert)
-          data = PulpCertguardClient::CertguardRHSMCertGuard.new(name: name, ca_certificate: ca_certificate)
+        def ca_cert
+          Cert::Certs.consumer_ca_cert
+        end
+
+        def refresh
+          found = list(name: 'RHSMCertGuard').results.first
+          if found
+            partial_update(found.pulp_href) if found.ca_certificate != ca_cert
+          else
+            create
+          end
+          persist_if_needed(found.pulp_href)
+        end
+
+        def persist_if_needed(href)
+          unless self.smart_proxy.pulp_mirror?
```

Comment:
```suggestion
          return if self.smart_proxy.pulp_mirror?
```

Could do a guard clause here to reduce the amount of nesting.